kernel/src/thread/mod.rs

```diff
@@ -77,52 +85,72 @@ impl Thread {
 
     /// Runs this thread at once.
     pub fn run(&self) {
-        self.set_status(ThreadStatus::Running);
+        self.status.store(ThreadStatus::Running, Ordering::Release);
         self.task.upgrade().unwrap().run();
     }
 
-    pub(super) fn exit(&self) {
-        self.set_status(ThreadStatus::Exited);
+    /// Returns whether the thread is exited.
+    pub fn is_exited(&self) -> bool {
+        self.status.load(Ordering::Acquire).is_exited()
     }
 
-    /// Returns the reference to the atomic status.
-    pub fn atomic_status(&self) -> &AtomicThreadStatus {
-        &self.status
+    /// Returns whether the thread is stopped.
+    pub fn is_stopped(&self) -> bool {
+        self.status.load(Ordering::Acquire).is_stopped()
     }
 
-    /// Returns the current status.
-    pub fn status(&self) -> ThreadStatus {
-        self.status.load(Ordering::Acquire)
+    /// Stops the thread if it is running.
+    ///
+    /// If the previous status is not [`ThreadStatus::Running`], this function
+    /// returns [`None`]. Otherwise, it sets the status to
+    /// [`ThreadStatus::Stopped`] and returns [`Some(())`].
+    ///
+    /// This function only sets the status to [`ThreadStatus::Stopped`],
+    /// without initiating a reschedule.
+    pub fn stop(&self) -> Option<()> {
```

Comment:
I wonder why we prefer `Option` to `Result` here, We may offer the caller more information like what the status before is with a `Result`. Same for the other status transition methods.